.github/workflows/ci-tests.yml

```diff
@@ -26,7 +26,13 @@ jobs:
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          pip install -e ".[tests]"
+          pip install poetry
+          pip install poetry-dynamic-versioning
+          pip install flake8
+          pip install mypy
+          pip install yapf
+          pip install toml
+          poetry install
```

Comment:
Poetry should install the dev dependencies as specified in our pyproject.toml, so we should not need to duplicate them here

Reply:
Unfortunately the linter Github action doesn't run from within the poetry environment, so we need to install these dependencies separately.

Reply:
Can this be rectified with the `poetry shell` command?

Reply:
No, shell doesn't work on Github's CI/CD servers.

Reply:
It looks like `source .venv/bin/activate` is an alternative (from the README of https://github.com/snok/install-poetry )